logstage/logstage-config/src/main/scala/com/github/pshirshov/izumi/logstage/config/codecs/RenderingPolicyCodec.scala

```diff
@@ -0,0 +1,102 @@
+package com.github.pshirshov.izumi.logstage.config.codecs
+
+import com.github.pshirshov.izumi.distage.config.codec.ConfigReader
+import com.github.pshirshov.izumi.distage.model.reflection.universe.RuntimeDIUniverse.u
+import com.github.pshirshov.izumi.logstage.api.rendering.RenderingPolicy
+import com.github.pshirshov.izumi.logstage.config.codecs.RenderingPolicyCodec.{NamedRenderingPolicy, RenderingPolicyMapper}
+import com.typesafe.config.{Config, ConfigFactory, ConfigValue}
+import RenderingPolicyCodec._
+
+import scala.util.Try
+
+class RenderingPolicyCodec(policyMappers: Set[RenderingPolicyMapper[_ <: RenderingPolicy]]) extends ConfigReader[RenderingPolicy] {
```

Comment:
If you make RenderingPolicyMapper covariant, you can remove the type bound to just `Set[RenderingPolicyMapper[RenderingPolicy]]` or just `Set[RenderingPolicyMapper[_]]`:

```scala
case class Str[+S <: String](s: S)
// if S is not covariant +S - it doesn't work

val xz = "xz"
val five = "5"

def notEqualTypes[X, Y](implicit ev: X =:= Y = null) = ev == null

notEqualTypes[five.type, String]
// res0: Boolean = true

val set: Set[Str[String]] = Set(
  Str("Just string"): Str[String]
  , Str(five: five.type): Str[five.type]
  , Str(xz: xz.type): Str[xz.type]
)
// set: Set[StringOption[String]] = Set(StringOption(None), StringOption(Some(5)), StringOption(Some(z)))
```